### PR TITLE
docs: Lending oracles

### PR DESCRIPTION
**Pull request by mo-anon**
add overview section and:
- `CryptoFromPool.vy`
- `CryptoFromPoolsRate.vy`

resolves https://github.com/CurveDocs/curve-docs/issues/164

### CODE REVIEW

File: docs/lending/contracts/cryptofrompoolsrate.md [hunk 1]
```diff
@@ -0,0 +1,620 @@
...
+    $\text{price} = 0.05 \times 1000 = 50$
+
+    This calculation indicates that one Token is equivalent to 50 crvUSD.
+
+
+
+### `price`
+!!! description "`CryptoFromPoolsRate.price() -> uint256`"
+
+    Getter for the price of the collateral denomintaed against the borrowed token. E.g. a market with pufETH as collateral and crvUSD borrowable, the price will return the pufETH price with regard to crvUSD.
```
Comment (saint-rat) [inline docs/lending/contracts/cryptofrompoolsrate.md:102]: >denomintaed

spelling

File: docs/lending/contracts/cryptofrompoolsrate.md [hunk 2]
```diff
@@ -0,0 +1,620 @@
...
+            ```
+
+    === "Example"
+        ```shell
+        >>> CryptoFromPoolsRate.cached_rate()
+        865388487987562874
+        ```
+
+
+### `chached_timestamp`
```
Comment (saint-rat) [inline docs/lending/contracts/cryptofrompoolsrate.md:415]: >chached_timestamp

spelling

File: docs/lending/contracts/cryptofrompoolsrate.md [hunk 3]
```diff
@@ -0,0 +1,620 @@
...
+                rate, use_rates = self._stored_rate()
+                if use_rates:
+                    self.cached_rate = rate
+                    self.cached_timestamp = block.timestamp
+                return rate
+            ```
+
+    === "Example"
+        ```shell
+        >>> CryptoFromPoolsRate.chached_timestamp()
```
Comment (saint-rat) [inline docs/lending/contracts/cryptofrompoolsrate.md:446]: >chached_timestamp

spelling again

File: docs/lending/contracts/cryptofrompoolsrate.md [hunk 4]
```diff
@@ -0,0 +1,620 @@
...
+
+        >>> CryptoFromPoolsRate.BORROWED_IX(1)
+        0                   # wstETH
+        ```
+
+
+### `COLLATERAL_IX`
+!!! description "`CryptoFromPoolsRate.COLLATERAL_IX(arg0: uint256) -> uint256: view`"
+
+    Getter for the index of the collateral token within the pool. TODO
```
Comment (saint-rat) [inline docs/lending/contracts/cryptofrompoolsrate.md:517]: > TODO

this seems finished to me?

**Review by saint-rat — CHANGES_REQUESTED** [review]
just a couple spellings really.

File: docs/lending/contracts/oracle-overview.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    ---
+
+    Oracle for collateral tokens using multiple different Curve pool oracles chained together. This oracle contract can also make use of rates.
+
+    [:octicons-arrow-right-24: `CryptoFromPoolsRate.vy`](./cryptofrompoolsrate.md)
+
+</div>
+
+!!!warning "Contract Immutability"
+    The oracle contracts are fully immutable. Once deployed, there are no function to stop it, change the sources form where if fetched the price or other things.
```
Comment (saint-rat) [inline docs/lending/contracts/oracle-overview.md:31]: >change the sources form where if fetched the price or other things

needs rewriting

File: docs/lending/contracts/oracle-overview.md [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+
+*This final value represents the price of FXN in terms of crvUSD by chaining together two oracles. All values are based on a scale of 1e18; hence, 129899651623057139817 would approximate to 129.71 crvUSD per FXN token.*
+
+
+---
+
+
+### **Chained Oracles with Rates (pufETH)**
+
+The [oracle contract](https://etherscan.io/address/0xb08eB288C57a37bC82238168ad96e15975602cd9) for the pufETH lending market integrates two Curve pool oracles and applies the `stored_rates` from the [pufETH/wstETH pool](https://etherscan.io/address/0xeeda34a377dd0ca676b9511ee1324974fa8d980d) due to the nature of the tokens.
```
Comment (saint-rat) [inline docs/lending/contracts/oracle-overview.md:88]: Could you add something explaining what rates are a little more?  I'm a bit confused what the stored rates actually are.  Do the stored rates only matter when they are stableswap pools?

**Review by saint-rat — CHANGES_REQUESTED** [review]

File: docs/lending/contracts/oracle-overview.md (hunk 6 above)
Comment (mo-anon) [reply docs/lending/contracts/oracle-overview.md:88]: yeah, will add some more

**Pull request merged**